configurator.go

```diff
@@ -32,7 +31,10 @@ func NewDefaultConfigurator(providers ...provider.Provider) Configurator {
 }
 
 func NewConfigurator(providers ...provider.Provider) Configurator {
-	return &configurator{providers: providers, factory: &defaultSetterFactory{}}
+	return &configurator{
+		providers: providers,
+		factory:   &setter.DefaultSetterFactory{},
```

Comment:
prefer creating an initializer in the setter pacakge i.e setter.NewDefaultSetterFactor()